src/peergos/shared/social/HttpSocialNetwork.java

```diff
@@ -40,62 +40,37 @@ private static String getProxyUrlPrefix(Multihash targetId) {
-            dout.flush();
-
-            return poster.postUnzip(urlPrefix + "social/followRequest", bout.toByteArray()).thenApply(res -> {
-                DataInputStream din = new DataInputStream(new ByteArrayInputStream(res));
-                try {
-                    return din.readBoolean();
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
-            });
-        } catch (IOException ioe) {
-            LOG.log(Level.WARNING, ioe.getMessage(), ioe);
-            return CompletableFuture.completedFuture(false);
-        }
+        return poster.postUnzip(urlPrefix + "social/followRequest?owner=" + encode(target.toString()), encryptedPermission).thenApply(res -> {
+            DataInputStream din = new DataInputStream(new ByteArrayInputStream(res));
+            try {
+                return din.readBoolean();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
     }
 
     @Override
-    public CompletableFuture<byte[]> getFollowRequests(PublicKeyHash owner) {
-        return getFollowRequests("", direct, owner);
+    public CompletableFuture<byte[]> getFollowRequests(PublicKeyHash owner, byte[] signedTime) {
+        return getFollowRequests("", direct, owner, signedTime);
     }
 
     @Override
-    public CompletableFuture<byte[]> getFollowRequests(Multihash targetServerId, PublicKeyHash owner) {
-        return getFollowRequests(getProxyUrlPrefix(targetServerId), p2p, owner);
+    public CompletableFuture<byte[]> getFollowRequests(Multihash targetServerId, PublicKeyHash owner, byte[] signedTime) {
+        return getFollowRequests(getProxyUrlPrefix(targetServerId), p2p, owner, signedTime);
     }
 
-    private CompletableFuture<byte[]> getFollowRequests(String urlPrefix, HttpPoster poster, PublicKeyHash owner)
+    private CompletableFuture<byte[]> getFollowRequests(String urlPrefix, HttpPoster poster, PublicKeyHash owner, byte[] signedTime)
     {
-        try
-        {
-            ByteArrayOutputStream bout = new ByteArrayOutputStream();
-            DataOutputStream dout = new DataOutputStream(bout);
-
-
-            Serialize.serialize(owner.serialize(), dout);
-            dout.flush();
-
-            return poster.postUnzip(urlPrefix + "social/getFollowRequests", bout.toByteArray()).thenApply(res -> {
-                DataInputStream din = new DataInputStream(new ByteArrayInputStream(res));
-                try {
-                    return CoreNodeUtils.deserializeByteArray(din);
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
-            });
-        } catch (IOException ioe) {
-            LOG.log(Level.WARNING, ioe.getMessage(), ioe);
-            return null;
-        }
+        return poster.get(urlPrefix + "social/getFollowRequests?owner=" + encode(owner.toString())
```

Comment:
what happens if this throws an  exception?

Reply:
Then you get an exceptionally completed future. (Any method that returns a future is expected not to throw otherwise you are mixing sync and async programming)